app/src/main/java/com/joseg/fakeyouclient/common/Result.kt

```diff
@@ -1,7 +1,27 @@
 package com.joseg.fakeyouclient.common
 
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onStart
+
 sealed interface Result<out T> {
     class Success<T>(val data: T) : Result<T>
-    class Error(val message: String) : Result<Nothing>
+    class Error(val exception: Throwable? = null) : Result<Nothing>
```

Comment:
Change the argument of `Result.Error(val message: String)` to receive a `Throwable` object instead.